src/controller/java/AndroidCallbacks.cpp

```diff
@@ -273,25 +267,40 @@ void ReportCallback::OnAttributeData(const app::ConcreteDataAttributePath & aPat
                    ReportError(attributePathObj, nullptr, CHIP_JNI_ERROR_EXCEPTION_THROWN));
 #endif
     // Create TLV byte array to pass to Java layer
+    TLV::TLVReader readerForJavaTLV;
+    readerForJavaTLV.Init(*apData);
     size_t bufferLen                  = readerForJavaTLV.GetRemainingLength() + readerForJavaTLV.GetLengthRead();
     std::unique_ptr<uint8_t[]> buffer = std::unique_ptr<uint8_t[]>(new uint8_t[bufferLen]);
     uint32_t size                     = 0;
-    // The TLVReader's read head is not pointing to the first element in the container, instead of the container itself, use
-    // a TLVWriter to get a TLV with a normalized TLV buffer (Wrapped with an anonymous tag, no extra "end of container" tag
-    // at the end.)
+
     TLV::TLVWriter writer;
     writer.Init(buffer.get(), bufferLen);
-    err = writer.CopyElement(TLV::AnonymousTag(), readerForJavaTLV);
+    TLV::TLVType outer;
+
+    err = writer.StartContainer(TLV::AnonymousTag(), TLV::kTLVType_Structure, outer);
```

Comment:
Will this change the existing TLV format we sent to application? It looks all attributes will be put under a struct with this operation.

I see the json format is still not within the struct 
attribute != null && attribute.getJson().toString().equals("""{"16:BOOL":false}""")))

It loosk the TLV format and Json format is not consistent

Reply:
"""{"16:BOOL":false}""") is structure. thanks

Reply:
This is current TLV format I  passed to app, 
{"0:UINT":0,"1:UINT":0,"2:UINT":0,"3:BOOL":false,"4:UINT":0,"65528:ARRAY-UINT":[0,1,2,3,4,6],"65529:ARRAY-UINT":[0,1,2,3,4,5,6],"65531:ARRAY-UINT":[65528,65529,65531,65532,65533,0,1,2,3,4],"65532:UINT":0,"65533:UINT":4}

After this change, I think it will changed to 
{{"0:UINT":0,} {"1:UINT":0}....}

Reply:
I think kTLVType_Structure only make sense if we need to pass more one type-value pairs in a single shot,  for attribute, this AnonymousTag() with Struct is extra, since we always need to unwrap it to get the type-value pair

Reply:
per offline chat, we go with structure, for a list of attribute, you just need to combine a list of json objects together.  I have added another function to show how to do that, thanks